src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -0,0 +1,200 @@
+package io.pinecone.helpers;
+
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.exceptions.PineconeException;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import org.openapitools.client.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.pinecone.helpers.BuildUpsertRequest.buildRequiredUpsertRequestByDimension;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class TestIndexResourcesManager {
+    private static final Logger logger = LoggerFactory.getLogger(IndexManager.class);
+    private static TestIndexResourcesManager instance;
+    private static final String apiKey = System.getenv("PINECONE_API_KEY");
+    private final int dimension = System.getenv("DIMENSION") == null
```

Comment:
These env vars are all stored in Github Secrets (or whatever it's called) when these tests are run online, right?

Reply:
Yeah, I was trying to provide some flexibility in how we set them. So we can set them in CI and drive different variations of testing. Jen did a really good job of writing the integration tests in this way in the Python client, and I wanted to give us the option here.

We mostly rely on the defaults for now.

Reply:
Awesome!